cvmfs/catalog_mgr_rw.cc

```diff
@@ -837,6 +847,34 @@ shash::Any WritableCatalogManager::SnapshotCatalog(WritableCatalog *catalog)
   return hash_catalog;
 }
 
+void WritableCatalogManager::Balance() {
+  CatalogList catalog_list = GetCatalogs();
+  for (unsigned i = 0; i < catalog_list.size(); ++i) {
+    FixWeight(static_cast<WritableCatalog*>(catalog_list[i]));
+  }
+}
+
+void WritableCatalogManager::FixWeight(WritableCatalog* catalog) {
+  // firstly check underflow because they can provoke overflows
+  if (catalog->GetNumEntries() < min_weight_ &&
+      !catalog->IsRoot() &&
+      catalog->IsAutogenerated()) {
+    LogCvmfs(kLogCatalog, kLogStdout,
+             "deleting an autogenerated catalog in '%s'",
+             catalog->path().c_str());
+    // Remove the .cvmfscatalog and .cvmfsautocatalog files first
+    string path = catalog->path().ToString();
+    catalog->RemoveEntry(path + "/.cvmfscatalog");
```

Comment:
It would be nicer without magic strings, yes.  A separate file might be the right think if the magic strings are used in multiple classes, otherwise a constant member would work, too.  This requires a record of the status quo first (what are our magic strings and where).